Feat Bootstrap and Fix status bug

SanGeeky opened the pull request:
In this feature, I add bootstrap with the help of webpack, and now the views have a new style. Also, I fix the bug in the status of an article.

Important changes:

1. Add dependencies to webpack in `environment.js`
2. Import bootstrap in `application.js`
3. Add `application.sccs` and import bootstrap in views
4. Style of views updated with a bootstrap template
5. Add `private?` method on concern `visible.rb`
6. Validate status in views

SanGeeky committed:
Feat: Add Bootstrap and modify webpack enviroment

Rails 6 uses webpack so, only is necessary install bootstrap
with yarn, and add it to webpack enviroment. Also is created
a new asset called application.scss that import bootstrap in
all views.

SanGeeky committed:
Fix webpack version 4.x.x and environment settings

SanGeeky committed:
Feat new bootstrap index for articles

SanGeeky committed:
Add links to root path

SanGeeky committed:
Remove comments in files.

SanGeeky committed:
Fix url for articles

SanGeeky committed:
Fix show article depending on its status

SanGeeky committed:
Reformat enviroment.js

SanGeeky committed:
Fix import css in application.scss

SanGeeky committed:
New Layout to all views.

SanGeeky committed:
Update documentation for bootstrap

SanGeeky committed:
Add target in footer icons

mauriciosabogal commented on package.json:
```diff
@@ -3,11 +3,15 @@
     "babel-loader": "^8.2.2",
+    "bootstrap": "^4.6.0",
+    "jquery": "^3.6.0",
+    "popper.js": "^1.16.1",
```
Why do you need to include` popper.js`?

SanGeeky replied:
According to Bootstrap documentation is required but I read about the usage of `popper.js` and only is required mainly for tooltips, popovers, dropdowns, nor modals. So I removed this dependency because I don't use those features, and works fine.

mauriciosabogal commented on app/assets/stylesheets/application.scss:
```diff
@@ -0,0 +1,4 @@
+@import 'bootstrap/scss/bootstrap';
+@import url('https://fonts.googleapis.com/css?family=Lora:400,700,400italic,700italic');
+@import url('https://fonts.googleapis.com/css?family=Open+Sans:300italic,400italic,600italic,700italic,800italic,400,300,600,700,800');
+@import 'clean-blog';
```
Why do you need to import `clean-blog` here?

SanGeeky replied:
I removed the `clean-blog` file and use the corresponding styles for each view.

mauriciosabogal commented on app/assets/stylesheets/clean-blog.css:
```diff
@@ -0,0 +1,410 @@
+/*!
```
Do you really need all of these css rules?

SanGeeky replied:
I removed some css rules, but I included the majority to give best style to buttons, text area, inputs, header, and navbar. And also these rules handle the style for small screens.

mauriciosabogal commented on app/views/articles/_form.html.erb:
```diff
@@ -1,23 +1,46 @@
+      </div>
+    </div>
+</div>
+
```
Please delete the extra empty lines.

SanGeeky replied:
Done

mauriciosabogal commented on app/views/articles/_form.html.erb:
```diff
@@ -1,23 +1,46 @@
-    <%= form.submit %>
-  </div>
-<% end %>
+<div class="container">
```
Please review the indentation of this file.

SanGeeky replied:
Done !

mauriciosabogal commented on app/views/articles/edit.html.erb:
```diff
@@ -1,2 +1,16 @@
+  </div>
+</header>
 <%= render 'form', article: @article %>
+
```
Please delete the extra line at the end of the file.

SanGeeky replied:
Done!

mauriciosabogal commented on app/views/articles/index.html.erb:
```diff
@@ -1,13 +1,46 @@
+
+
+
+
```
Please delete the extra lines at the end of the file.

SanGeeky replied:
Done!

mauriciosabogal commented on app/views/articles/index.html.erb:
```diff
@@ -1,13 +1,46 @@
-  <% end %>
-</ul>
-<%= link_to 'New Article', new_article_path %>
+
```
Please review the indentation in this file.

SanGeeky replied:
Done !

mauriciosabogal commented on app/views/articles/show.html.erb:
```diff
@@ -1,16 +1,49 @@
-  <li><%= link_to 'Edit', edit_article_path(@article) %></li>
-  <li>
-    <%= link_to 'Destroy',
+
```
Please review the indentation of this file.

SanGeeky replied:
Done!

mauriciosabogal commented on app/views/articles/_form.html.erb:
```diff
@@ -1,23 +1,46 @@
+          </div>
+          <br>
+          <div>
+            <%= form.submit 'Send', :class => "btn btn-outline-primary"%>
```
Remember to this syntax `class: 'btn btn-outline-primary'`.

SanGeeky replied:
Thanks !

mauriciosabogal commented on app/views/articles/show.html.erb:
```diff
@@ -1,16 +1,49 @@
+        <div class="d-grid gap-2 col-12 mx-auto">
+        <%= link_to 'Edit',
+                    edit_article_path(@article),
+                    :class => "btn btn-outline-primary"%>
```
`class: btn btn-outline-primary'`

SanGeeky replied:
Thanks

mauriciosabogal commented on app/views/comments/_form.html.erb:
```diff
@@ -1,17 +1,28 @@
+  </div>
+  <br>
+  <div class="control-group">
+    <%= form.submit 'Send', :class => "btn btn-outline-dark"%>
```
`class: 'btn btn-outline-dark'`

SanGeeky replied:
Done!

mauriciosabogal commented on app/views/layouts/application.html.erb:
```diff
@@ -9,8 +9,66 @@
     <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
     <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
   </head>
+    <!-- Navigation -->
```
Please review the indentation of this file.

SanGeeky replied:
Done !

mauriciosabogal replied:
Please review the indentation of this file again.

mauriciosabogal reviewed (changes requested):
Please review the ruby code on the .erb files. Apply coding practices from rubocop.

mauriciosabogal commented on app/views/articles/edit.html.erb:
```diff
@@ -1,2 +1,16 @@
-<h1>Edit Article</h1>
+<!-- Page Header -->
+<header class="masthead" style="background-image: url(<%= asset_path 'create-bg.jpg' %>);">
```
You are using a `header` in several pages for the articles. How can you refactor the code to avoid this repetition?

SanGeeky replied:
With `yield` and `content_for` I refactor the `header` and modify the necessary text in each view.

SanGeeky committed:
Remove popper.js dependency

SanGeeky committed:
Refactor code in views

1. Header in Application
2. Remove spaces
3. Fix Identation

SanGeeky committed:
Refactor clean-blog styles

1.  application.scss contains necesary styles in all views
2. articles.scss contains styles for show an article
3. forms.scss is used for comments an articles forms
4. yield in application.html.erb to import the corresponding style

mauriciosabogal commented on README.md:
```diff
@@ -158,3 +158,76 @@ https://sangeekyonrails.herokuapp.com/
 - [Heroku Puma documentation](https://devcenter.heroku.com/articles/deploying-rails-applications-with-the-puma-web-server#adding-puma-to-your-application)
+
+
+# Add Bootstrap in Rails 6
```
Please update the README. We are not using `popper.js` anymore.

SanGeeky committed:
Format erb files and apply rubocop style guide.

SanGeeky committed:
Refactor stylesheet tag in views

SanGeeky committed:
Update bootstrap documentation

SanGeeky committed:
Refactor stylesheets assets with require_tree
